kcidb/test_db.py

```diff
@@ -19,25 +20,19 @@ def test_schemas_main():
                     stdout_re=r"4\.0: 4\.0\n4\.1: 4\.1\n")
 
 
-def test_init_main():
+def test_init_main(empty_database):
     """Check kcidb-db-init works"""
     argv = ["kcidb.db.init_main", "-d", "bigquery:project.dataset"]
     driver_source = textwrap.dedent("""
-        from unittest.mock import patch, Mock
-        client = Mock()
-        client.init = Mock()
-        client.is_initialized = Mock(return_value=False)
-        with patch("kcidb.db.Client", return_value=client) as \
-                Client:
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        client.init.assert_called_once()
+        if not empty_database:
+            pytest.skip("Empty database not configured")
+        status = kcidb.db.init_main()
```

Comment:
If the pytest configuration doesn't have the fixture available, then the test won't be executed. We don't need to check for it here.

There is a bit of a problem before we can run those tests with these fixtures, actually.

The entry points for the command-line tools we're testing here (i.e. `kcidb.db.init_main`, which is an entry point for the `kcidb-db-init` tool) expect a database specification string (run `kcidb-db-init --database-help` to get a description of those), while the `empty_database` and `clean_database` fixtures are objects of type `kcidb.db.Client` representing an already-established connection to a database. One cannot be used in place of another as is. However, the `kcidb.db.Client` objects are created with a database specification string as the input. You could modify that class to store the specification string, say in `self.database`, and then extract it from the `empty_database` or `clean_database` fixture in the test, to pass to the tool's entry point. I don't think you would need to pass **any** driver_source to `assert_executes` here, as we're not going to be doing any "mocking".

If you'd like we can have a call tomorrow, and I'll walk you through what's going on. Reach out on Slack and tell me which times you have available, if you'd like to do that.

Reply:
implemented changes @spbnick